src/components/taskDetails/index.tsx

```diff
@@ -0,0 +1,201 @@
+import React, { ChangeEvent, FC } from 'react';
+import NavBar from '@/components/navBar/index';
+import TaskContainer from './TaskContainer';
+import Details from './Details';
+import classNames from './task-details.module.scss';
+import { toast, ToastTypes } from '@/helperFunctions/toast';
+import task from '@/interfaces/task.type';
+import { useState, useEffect, useContext } from 'react';
+import useFetch from '@/hooks/useFetch';
+import { isUserAuthorizedContext } from '@/context/isUserAuthorized';
+import fetch from '@/helperFunctions/fetch';
+
+type Props = {
+  url: string;
+  taskID: string;
+};
+
+const TaskDetails: FC<Props> = ({ url, taskID }) => {
+  const isAuthorized = useContext(isUserAuthorizedContext);
+  const [isEditing, setIsEditing] = useState(false);
+  const [initialData, setInitialData] = useState<task>();
+  const [taskDetails, setTaskDetails] = useState<task>();
+  const [editedDetails, setEditedDetials] = useState({});
+  const { response, error, isLoading } = useFetch(url);
+  const { SUCCESS, ERROR } = ToastTypes;
+
+  useEffect(() => {
+    const fetchedData: task = { ...response.taskData };
+    setTaskDetails(fetchedData);
+    setInitialData(fetchedData);
```

Comment:
why are we setting both initial and task detail? Doesn't `initial` specify the initial state?

Reply:
Initially, all the data will be fetched and stored in both `taskDetails` and `initialData`. After that, we will use' taskDetails' to load the data on UI. When the superuser clicks on the `Edit` button he/she gets two options `Save` or `Cancel`. In the case of cancellation we'll need this `initialData` to restore the data back on UI.

Reply:
resolved